docs/design/storage-auto-scale.md

```diff
@@ -72,35 +72,53 @@ Controller:
 
 1) Create a named controller that watches for [channel generic event](https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches) per device class.
 
-2) If the expansion is not in progress set the status `phase` to `NotStarted` 
+2) Set the status `phase` to `NotStarted`, if no expansion has triggered(status.phase==""). 
+   
+3) Check if the expansion is in progress:
+   
+   1) Query actualOsdCount and actualOsdSize from the from Prometheus .
+   
+   2) Calculate the desiredOsdCount and desiredOsdSize from the Storagecluster.
+   
+   3) If the (actualOsdSize < desiredOsdSize || actualOsdCount < desiredOsdCount), expansion is in progress. 
    
-3) If an expansion is in progress(expectedOsdSize!=startOsdSize || expectedOsdCount!=startOsdCount), check the progress and then requeue each 1 minute until the expansion is completed successfully(jump to step 11).
+4) If the expansion is in progress, check the progress and then requeue each 1 minute until the expansion is completed successfully(jump to step 13).
    
-4) If the LSO storageclass is detected in the storageClassDeviceSet, raise a warning and do not recocnile further.
+5) If no-expansion is in progress,
    
-5) If the highest osd percentage reported in the sync map is more than osdScalingThresholdPercent(70%) means reaching osd nearfull, scaling is needed.
+   1) Check the (status.phase == "InProgress"), if yes move the status phase to succeeded.
```

Comment:
If the phase is still in progress, shouldn't we just to step 13? It seems odd that it's still in progress, but we will evaluate whether to again expand it in the same reconcile.

Reply:
I think we don't even need step 13, as we are doing same at step 3)

What if we remove step 13) and keep the step 5) as it is